plugins/module_utils/api.py

```diff
@@ -262,6 +264,26 @@ def create(self, resource, data=None):
             resource['state'] = "present"
         return resource
 
+    def wait_for_state(self, check_parameter, allowed_states):
+        start = datetime.now()
+        timeout = self._module.params['api_timeout'] * 2
+        while datetime.now() - start < timedelta(seconds=timeout):
+            info = self.query()
+            if info.get(check_parameter) in allowed_states:
+                return info
+            sleep(1)
+
+        # Timeout succeeded
+        name_uuid = info.get('name') or self._module.params.get('name') or \
+            self._module.params.get('uuid')
+
+        if name_uuid is not None:
+            msg = "Timeout while waiting for a state change for resource %s to states %s" % (name_uuid, allowed_states)
+        else:
+            msg = "Timeout while waiting to find resource %s" % name_uuid
```

Comment:
I don't think `name_uuid` can be `None`. You either need a name or a UUID. And even if it's None under special circumstances I don't think this if/else makes much sense:
 ```suggestion
        msg = "Timeout while waiting for a state change for resource %s to states %s" % (name_uuid, allowed_states)
```